tests/unit_tests/fn/test_fn_scan.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include <gridtools/fn/scan.hpp>
+
+#include <gtest/gtest.h>
+
+#include <gridtools/sid/composite.hpp>
+#include <gridtools/sid/synthetic.hpp>
+
+namespace gridtools::fn {
+    namespace {
+        using namespace literals;
+        using sid::property;
+
+        struct sum_fold : fwd {
+            static GT_FUNCTION constexpr auto body() {
+                return [](auto acc, auto const &iter) { return acc + *iter; };
+            }
+        };
+
+        struct sum_scan : fwd {
+            static GT_FUNCTION constexpr auto body() {
+                return scan_pass(
+                    [](auto acc, auto const &iter) {
+                        return tuple_util::host_device::make<tuple>(get<0>(acc) + *iter, get<1>(acc) * *iter);
+                    },
+                    [](auto acc) { return get<0>(acc); });
```

Comment:
it is strange that unqualified `get<0>` is compiled here. I am not even able to get what overload is chosen.

Reply:
Why? It’s just `gridools::get` defined in `common/tuple.hpp`, isn’t it?

Reply:
yep. You are right.